### DISCUSSION TITLE
Multiline environment variables

### DISCUSSION THREAD

**Discussion by dawnerd** [vercel/next.js#37935] [Help]
Just wondering if we can get dotenv updated in @next/env. They added support a while ago but looks like next is pinned to a fairly old version. https://github.com/motdotla/dotenv/commit/0a9332ff70f5121c1221d5c32839a759c60cb879

My use case is setting a private key (for generating an apple mapkit token). Seems the only solution right now is to either hardcode the key in the file as a temp measure or load the file from code. Either case ends up with the key being stored in the repo which I'm not a huge fan of.

**Comment by motdotla** [comment]
Hi @dawnerd - creator of dotenv and dotenv-expand here 👋

I believe you could do this as a workaround until Next.js makes this update:

1. Go to [https://github.com/dotenv-org/integration-example-vercel-edge-function](https://github.com/dotenv-org/integration-example-vercel-edge-function) for an example
2. Add dotenv to your package.json - `yarn add dotenv`
3. Modify your next.config.js like this:

```js
const result = require('dotenv').config()

const nextConfig = {
  reactStrictMode: true,
  env: result.parsed,
}

module.exports = nextConfig
```

**Reply by motdotla** [reply]
P.S. Your opencoaster project looks neat! 🎢

**Reply by dawnerd** [reply]
💥 Thanks, that is actually a pretty nice workaround too!
